src/test/kotlin/de/smartsquare/starter/mqtt/MqttHandlerTest.kt

```diff
@@ -0,0 +1,214 @@
+package de.smartsquare.starter.mqtt
+
+import com.fasterxml.jackson.module.kotlin.jsonMapper
+import com.hivemq.client.mqtt.datatypes.MqttQos
+import com.hivemq.client.mqtt.mqtt5.message.publish.Mqtt5Publish
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+object TestMqttSubscriberCollector {
+    operator fun invoke(bean: Any): MqttSubscriberCollector = MqttSubscriberCollector(MqttProperties()).apply {
+        postProcessAfterInitialization(bean, "testBean")
+    }
+}
+
+@Suppress("RedundantSuspendModifier")
+class MqttHandlerTest {
```

Comment:
I always appreciate more tests, good work! These look very similar to the `MqttXAutoConfigurationTest` classes. What additional value do they bring? Can't we just extend those to cover the additional cases surrounding `suspend` functions?

Reply:
Your existing tests are integration tests, which are slow, as they start a dedicated container. I prefer having fast unit tests while developing. The test I wrote run within a few hundred ms on my machine. If you insist (you are somewhat right, that they do similar things), I could iterate my tests by extending the existing ones.

Reply:
Ah, so this is a question of personal preference. I like to have mostly integration tests since they test more real code and find more real bugs (from experience). Also, unit tests are usually more work intensive since they often need to be changed when the code under test changed. It's a trade-off in the end.

Since you already wrote these tests and they test new cases we can of course keep them. Adding a case for suspend function to the integrations would be appreciated though :slightly_smiling_face:!

Reply:
Okay I did add those 😄